### PR TITLE
Add API doc to secondary package entry points

### PR DESCRIPTION
**Pull request by jbogarthyde**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?
 API doc for package pages missing from sub-packages

Issue Number: N/A

## What is the new behavior?
Doc added to all package pages 
https://angular.io/api?type=package

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by jbogarthyde** [comment]
CARETAKER  - This touches API in all the code-owner areas, but the sub-package descriptions are short and straightforward - it wouldn't take long to check it.

File: packages/upgrade/PACKAGE.md [hunk 1]
```diff
@@ -1,2 +1,3 @@
-Supports the upgrade path from AngularJS to Angular, allowing
-components from both systems to be used in the same application.
\ No newline at end of file
+**Deprecated** Provided support for upgrading applications from Angular JS to Angular.
```
Comment (gkalpak) [inline packages/upgrade/PACKAGE.md]: This file pertains to the `@angular/upgrade` package. The package is not deprecated. (The primary `upgrade` entry-point is deprecated, which is different than deprecating the package.)

**Review by gkalpak — CHANGES_REQUESTED** [review]
~~I don't think your changes have the effect you expect. AFAICT, the `PACKAGE.md` files pertain to a package (as the name suggests). Packages are only the top-level directories under `packages/` (e.g. `packages/animations/`, `packages/core/`, etc.).~~

~~The sub-directories do not represent sub-packages; they are secondary entry-points (which is a different thing). Currently, I don't think we support a separate docs page for entry-points.~~

~~So, only the top-level packages should have `PACKAGE.md` files (but you may want to add info about the secondary entry-points there).~~

_EDIT: It turns out I was wrong. We do take `PACKAGE.md` into account for secondary entry-points._
_(My commend about `@angular/upgrade` not being deprecated is still valid though.)_

File: packages/common/PACKAGE.md [hunk 2]
```diff
@@ -1,3 +1,3 @@
-Implements basic Angular directives and pipes, such as NgIf, NgForOf, DecimalPipe, and so on. 
+Implements basic Angular directives and pipes, such as `NgIf`, `NgForOf`, `DecimalPipe`, and so on.
```
Comment (gkalpak) [inline packages/common/PACKAGE.md]: `@angular/common` does not only include directives and pipes, but some other things, such as `Location`, `LocationStrategy`, `ViewportScroller`, etc.

File: packages/animations/test/PACKAGE.md [hunk 3]
```diff
@@ -0,0 +1 @@
+Supplies a testing module for the Angular Animations driver.
```
Comment (gkalpak) [inline packages/animations/test/PACKAGE.md]: This directory is not a package or entry-point. (It just contains our automated tests.)
It should not have a `PACKAGE.md` file.

File: packages/common/testing/PACKAGE.md [hunk 4]
```diff
@@ -0,0 +1 @@
+Supplies a testing module for basic Angular directives.
```
Comment (gkalpak) [inline packages/common/testing/PACKAGE.md]: As mentioned in a previous comment, `@angular/common` does not only include directives and pipes. Also, this modules doesn't offer much in terms of testing directives/pipes (afaict). It mostly contains mock implementations for its services (e.g. `MockLocationStrategy`, `MockPlatformLocation, `SpyLocation`).

How about:
> Provides infrastructure for testing Angular common functionality.

or

> Provides infrastructure for testing functionality provided by the \`common\` package.

File: packages/upgrade/static/PACKAGE.md [hunk 5]
```diff
@@ -0,0 +1,2 @@
+Supports the upgrade path from AngularJS to Angular, allowing
+components from both systems to be used in the same application.
```
Comment (gkalpak) [inline packages/upgrade/static/PACKAGE.md]: Not only components, but also services.

**Review by gkalpak — CHANGES_REQUESTED** [review]

**Review by gkalpak — APPROVED** [review]

**Comment by IgorMinar** [comment]
@jbogarthyde @gkalpak  I just noticed this change, and realized that we are using the terminology wrong here.

most of the items in this PR are entry-points (https://angular.io/guide/glossary#entry-point) and not npm packages (https://angular.io/guide/glossary#npm-package).

e.g. @angular/common is package that contains several entry-points, including the primary entry-point @angular/common (the same name as the package), but also other entry-points like: @angular/common/testing, @angular/common/http, etc. If you go to https://angular.io/api/common you can see all the primary and secondary entrypoints and if you follow a link to one of the secondary entry point docs, you'll find it documented as a package (violet badge "PACKAGE") and not as an entry-point.

Is this a limitation of our docs-infra?

Ideally the docs-infra should enable us to document these entry points correctly, and not accept invalid package names like @angular/common/http.

@gkalpak or @jbogarthyde can you please add this to the docs-infra backlog?

**Comment by gkalpak** [comment]
@IgorMinar: Yes, it seems to be a docs-infra limitation. I guess we only supported packages at first and later we piggy-backed on that feature to also document secondary entry-points.

I created #34081 and added it to the backlog of the docs-infra GitHub project.

**Pull request closed without merging**